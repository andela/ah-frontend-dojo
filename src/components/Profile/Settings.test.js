import React from 'react';
import { shallow, mount } from 'enzyme';
import {
  Settings,
  mapDispatchToProps,
} from 'components/Profile/Settings';

describe('Setting Component', () => {
  const mockedFunction = jest.fn();
  const component = shallow(<Settings updateProfile={mockedFunction} username="nadralia" />);
  it('should match the snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('submit user details', () => {
    component.setProps({
      userData: { firstname: 'john', lastname: 'doe', bio: 'this is biodata' }
    });

    const event = {
      userData: { profile: { firstname: 'john', lastname: 'doe', bio: 'this is biodata' } },
      preventDefault: jest.fn()
    };

    const instance = component.instance();
    instance.handleSubmit(event);
    expect(instance.props.updateProfile).toBeCalled();
  });
  it('should handle the onchange event', () => {
    const event = {
      target: {
        name: 'firstname',
        value: 'nadralia',
      },
    };

    component.instance().handleChange(event);
    expect(component.instance().state.firstname).toBe('nadralia');
  });

  it('should map dispatch to props', () => {
    const mockedDispatch = jest.fn();

    mapDispatchToProps(mockedDispatch).updateProfile();
    expect(mockedDispatch).toHaveBeenCalled();
  });
});
