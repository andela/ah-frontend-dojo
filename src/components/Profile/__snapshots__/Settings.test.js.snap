// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Setting Component should match the snapshot 1`] = `
<div
  className="settings"
>
  <div
    className="settings__edit-profile"
  >
    <Form
      as="form"
      onSubmit={[Function]}
    >
      <div>
        <span
          className="image_label"
        >
           Upload a profile picture
        </span>
        <input
          accept="image/*"
          className="custom-file-upload"
          name="image"
          onChange={[Function]}
          placeholder="Select Profile Picture"
          type="file"
        />
      </div>
      <FormGroup
        widths="equal"
      >
        <FormInput
          as={[Function]}
          control={[Function]}
          defaultValue=""
          fluid={true}
          label="First name"
          name="firstname"
          onChange={[Function]}
          placeholder="John"
        />
        <FormInput
          as={[Function]}
          control={[Function]}
          defaultValue=""
          fluid={true}
          label="Last name"
          name="lastname"
          onChange={[Function]}
          placeholder="Doe"
        />
      </FormGroup>
      <FormTextArea
        as={[Function]}
        className="settings__edit-profile--textareastyled"
        control={[Function]}
        defaultValue=""
        label="Bio"
        maxLength="300"
        name="bio"
        onChange={[Function]}
        placeholder="Talk about yourself."
        rows={5}
      />
      <div
        className="field"
      >
        <Button
          btnClass="edit-btn"
          btnEvent={[Function]}
          btnName="Edit Profile"
          id="editProfile__button"
          type="submit"
        />
      </div>
    </Form>
  </div>
</div>
`;
